R/api_client.R

```diff
@@ -105,7 +105,46 @@ ApiClient  <- R6::R6Class(
 
     CallApi = function(url, method, queryParams, headerParams, body, ...){
 
-      # MANUAL EDIT AFTER OPENAPI AUTOGEN
+      resp <- self$ExecuteWrapped(url, method, queryParams, headerParams, body, ...)
+      statusCode <- httr::status_code(resp)
+
+      if (is.null(self$maxRetryAttempts)) {
+        self$maxRetryAttempts = 3
+      }
+      
+      if (!is.null(self$retryStatusCodes)) {
+        for (i in 1 : self$maxRetryAttempts) {
```

Comment:
Small nit mostly because I (just like every other user of R) have been bitten by this):  if the right-hand side of the colon is by accident zero, it will still enter the sequence running for i = 1 and then i = 0.  A better more defensice apprpach is to use `for (i in seq_len(self$maxRetryAttempts))` as `seq_len()` does The Right Thing (TM) here for loops:

```r
> seq_len(4)
[1] 1 2 3 4
> seq_len(0)
integer(0)
> seq_len(-1)
Error in seq_len(-1) : argument must be coercible to non-negative integer
> 
```

Reply:
Thanks @eddelbuettel -- I've been bitten by that as well!! :D